apps/vaporgui/hide_std_error_util.cpp

```diff
@@ -6,17 +6,28 @@
 #endif
 
 static int _savedSTDERR;
+static fpos_t pos;
 
 void HideSTDERR()
 {
 #ifndef WIN32
+    fflush( stderr );
+    fgetpos( stderr, &pos );
```

Comment:
Error checking is needed for any system call that returns an error status. It's not critical that we return an error message in the event of failure. So my recommendation would be to simply return from the function upon error, and set _savedSTDERR to an invalid value (e.g. -1). RestoreSTDERR() could then check _savedSTDERR before proceeding.